src/components/Layouts/Primary/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { ReactElement, useEffect } from 'react';
+import type { FC } from 'react';
+import {
+  Box,
+  Drawer,
+  Hidden,
+  List,
+  makeStyles,
+  Theme,
+} from '@material-ui/core';
+import { useRouter } from 'next/router';
+import NextLink from 'next/link';
+import { useTranslation } from 'react-i18next';
+import { NavItem } from './NavItem/NavItem';
+import logo from 'src/images/logo.svg';
+import { ReportNavItems } from 'src/components/Reports/NavReportsList/ReportNavItems';
+import { ToolsList } from 'src/components/Tool/Home/ToolList';
+
+interface NavBarProps {
+  accountListId: string;
+  onMobileClose: () => void;
+  openMobile: boolean;
+}
+
+interface Item {
+  href?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+  items?: Item[];
+  title: string;
+}
+
+interface Section {
+  as?: string;
+  href?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+  items?: Item[];
+  title: string;
+}
+
+const sections: Section[] = [
+  {
+    title: 'Dashboard',
+    href: '/accountLists/[accountListId]',
+  },
+  {
+    title: 'Contacts',
+    href: '/accountLists/[accountListId]/contacts',
+  },
+  {
+    title: 'Tasks',
+    href: '/accountLists/[accountListId]/tasks',
+  },
+  {
+    title: 'Reports',
+    items: ReportNavItems.map((item) => ({
+      ...item,
+      title: item.subTitle ? `${item.title} (${item.subTitle})` : item.title,
+      href: `/accountLists/[accountListId]/reports/${item.id}`,
+    })),
+  },
+  {
+    title: 'Tools',
+    items: ToolsList.flatMap((toolsGroup) => [
+      ...toolsGroup.items.map((tool) => ({
+        title: tool.tool,
+        href: `/accountLists/[accountListId]/reports/${tool.id}`,
+      })),
+    ]),
+  },
+  {
+    title: 'Coaches',
+    href: '/coaches',
+  },
+];
+
+function renderNavItems({
+  accountListId,
+  items,
+  pathname,
+  depth = 0,
+}: {
+  accountListId: string;
+  items: Item[];
+  pathname: string;
+  depth?: number;
+}) {
+  return (
+    <List disablePadding>
+      {items?.reduce<ReactElement[]>(
+        (acc, item) =>
+          reduceChildRoutes({
+            acc,
+            accountListId,
+            item,
+            pathname,
+            depth,
+          }),
+        [],
+      )}
+    </List>
+  );
+}
+
+function reduceChildRoutes({
+  acc,
+  accountListId,
+  pathname,
+  item,
+  depth,
+}: {
+  acc: ReactElement[];
+  accountListId: string;
+  pathname: string;
+  item: Item;
+  depth: number;
+}) {
+  const key = item.title + depth;
+  const { t } = useTranslation();
```

Comment:
I think this needs to be called in `NavBar` since it's a helper function. Although maybe you're treating it as a component...